common.go

```diff
@@ -41,7 +41,7 @@ const (
 //
 // The callback may be used for authentication/authorization and/or custom
 // transport wrapping.
-type OnConnectFunc func(remoteAddr string, rwc io.ReadWriteCloser) (io.ReadWriteCloser, error)
+type OnConnectFunc func(rwc net.Conn) (net.Conn, string, error)
```

Comment:
might worth to rename `rwc net.Conn` to `conn net.Conn` to keep it semantically consistent with other code